src/Strategy/UserSpecified/YamlDependencies.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.UserSpecified.YamlDependencies
+  ( analyze,
+    discover,
+    buildGraph,
+    UserDependencies (..),
+  )
+where
+
+import Control.Effect.Diagnostics
+import Data.Aeson
+import Data.Aeson.Types (Parser)
+import qualified Data.Map.Strict as M
+import Data.Text (Text, unpack)
+import Debug.Trace
+import DepTypes
+import Discovery.Walk
+import Effect.ReadFS
+import Graphing (Graphing)
+import qualified Graphing
+import Path
+import Types
+
+discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
+discover dir = map mkProject <$> findProjects dir
+
+-- Only search for fossa-deps.yaml in the root directory. We can extend this to subdirectories in the future.
+findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [UserDependenciesYamlProject]
+findProjects = walk' $ \dir _ files -> do
+  _ <- traceM $ show files
+  case findFileNamed "fossa-deps.yaml" files of
```

Comment:
I think `.yml` is much more common as an extension for yaml files; we should probably use it instead.